File: Codigos en C++/Ejercicio2_Ape06.cpp
#include <iostream> 
#include <vector> 
#include <algorithm> 
#include <numeric> 
#include <iomanip> 
using namespace std; 
int main() { 
vector<double> notas = {8.5, 7.0, 9.2, 6.4, 10.0, 5.8}; 
cout << "Notas registradas:" << endl; 
for (double nota : notas) 
cout << "  " << nota << endl;
double maximo = *max_element(notas.begin(), notas.end()); 
double minimo = *min_element(notas.begin(), notas.end()); 
double suma   = accumulate(notas.begin(), notas.end(), 0.0); 
double promedio = suma / notas.size(); 
cout << "Nota más alta : " << maximo << endl; 
cout << "Nota más baja : " << minimo << endl; 
cout << fixed << setprecision(2); 
cout << "Promedio      : " << promedio << endl; 
return 0; 
}         

File: Codigos en Java/Ejercicio2_Ape06.java
import java.util.ArrayList; 
import java.util.Collections; 
public class Ejercicio2_Ape06 { 
public static void main(String[] args) { 
ArrayList<Double> notas = new ArrayList<>(); 
notas.add(8.5);  notas.add(7.0);  notas.add(9.2); 
notas.add(6.4);  notas.add(10.0); notas.add(5.8); 
System.out.println("Notas registradas:"); 
for (double nota : notas) 
System.out.println("  " + nota); 
double maximo = Collections.max(notas); 
double minimo = Collections.min(notas); 
double suma = 0;
for (double n : notas) suma += n; 
double promedio = suma / notas.size(); 
System.out.println("Nota más alta : " + maximo); 
System.out.println("Nota más baja : " + minimo); 
System.out.printf("Promedio       : %.2f%n", promedio); 
} 
} 